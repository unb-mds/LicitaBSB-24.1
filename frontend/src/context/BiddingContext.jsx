import { createContext, useState } from 'react';
import { getLicitacoes } from '../services/licitacoes.service';

export const BiddingContext = createContext();
BiddingContext.displayName = 'Bidding';

export const BiddingProvider = ({ children }) => {
  const licitacoes = getLicitacoes();
  const [biddings, setBiddgins] = useState([...licitacoes]);
  const [words, setWords] = useState('');
  const [searchBiddings, setSearchBiddgins] = useState([]);
  return (
    <BiddingContext.Provider
      value={{
        biddings,
        setBiddgins,
        searchBiddings,
        setSearchBiddgins,
        words,
        setWords,
      }}
    >
      {children}
    </BiddingContext.Provider>
  );
};
