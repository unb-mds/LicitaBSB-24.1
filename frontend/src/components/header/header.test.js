import React from "react";
import '@testing-library/jest-dom'
import { render, screen } from "@testing-library/react";
import Header from ".";
import { MemoryRouter } from "react-router-dom";
import { BiddingProvider } from "../../context/BiddingContext";
import userEvent from '@testing-library/user-event'

describe("Deve renderizar os links das rotas: ", () => {
    test("Licitações", () => {
        render(<MemoryRouter>
            <BiddingProvider>
                <Header />
            </BiddingProvider>
        </MemoryRouter>)
        const linkRoute = screen.getAllByText('Licitações')
        expect(linkRoute).toHaveLength(2)
    })
    test("Sobre as Licitações", () => {
        render(<MemoryRouter>
            <BiddingProvider>
                <Header />
            </BiddingProvider>
        </MemoryRouter>)
        const linkRoute = screen.getAllByText('Sobre as Licitações')
        expect(linkRoute).toHaveLength(2)
    })
    test("Sobre o Projeto", () => {
        render(<MemoryRouter>
            <BiddingProvider>
                <Header />
            </BiddingProvider>
        </MemoryRouter>)
        const linkRoute = screen.getAllByText('Sobre o Projeto')
        expect(linkRoute).toHaveLength(2)
    })
    test("Sobre Nós", () => {
        render(<MemoryRouter>
            <BiddingProvider>
                <Header />
            </BiddingProvider>
        </MemoryRouter>)
        const linkRoute = screen.getAllByText('Sobre Nós')
        expect(linkRoute).toHaveLength(2)
    })
    test("Dashboard", () => {
        render(<MemoryRouter>
            <BiddingProvider>
                <Header />
            </BiddingProvider>
        </MemoryRouter>)
        const linkRoute = screen.getAllByText('Dashboard')
        expect(linkRoute).toHaveLength(2)
    })
})

test("Deve renderizar a lista de links presentes no componente", () => {
    render(<MemoryRouter>
        <BiddingProvider>
            <Header />
        </BiddingProvider>
    </MemoryRouter>)
    const listaDeLinks = screen.getAllByRole('listitem')
    expect(listaDeLinks).toHaveLength(10)
})

test("Deve renderizar o snapshot da lista de links do componente", () => {
    render(<MemoryRouter>
        <BiddingProvider>
            <Header />
        </BiddingProvider>
    </MemoryRouter>)
    const listaDeLinks = screen.getAllByRole('listitem')
    expect(listaDeLinks).toMatchSnapshot()
})

describe("Deve renderizar um campo de input", () => {
    test("no documento", () => {
        render(<MemoryRouter>
            <BiddingProvider>
                <Header />
            </BiddingProvider>
        </MemoryRouter>)
        const campoTexto = screen.getByPlaceholderText("Pesquise aqui")
        expect(campoTexto).toBeInTheDocument()
    })

    test("de type: text", () => {
        render(<MemoryRouter>
            <BiddingProvider>
                <Header />
            </BiddingProvider>
        </MemoryRouter>)
        const campoTexto = screen.getByPlaceholderText("Pesquise aqui")
        expect(campoTexto).toBeInTheDocument('type', 'text')
    })

    test(" de type: text que pode ser preenchido", async () => {
        render(
            <MemoryRouter>
                <BiddingProvider>
                    <Header />
                </BiddingProvider>
            </MemoryRouter>
        );
        const campoTexto = screen.getByPlaceholderText("Pesquise aqui");
        await userEvent.type(campoTexto, 'inputtest');
        expect(campoTexto).toHaveValue('inputtest');
    });
})