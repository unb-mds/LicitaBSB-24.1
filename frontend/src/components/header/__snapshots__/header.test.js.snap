// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o snapshot da lista de links do componente 1`] = `
Array [
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/licitacoes"
    >
      Licitações
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/SobreLicitacao"
    >
      Sobre as Licitações
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/sobredispensadelicitacao"
    >
      Sobre Dispensas
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/sobrebot"
    >
      Sobre o Bot
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/sobrelisitaBSB"
    >
      Sobre o Projeto
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/sobrenos"
    >
      Sobre Nós
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/dashboard"
    >
      Dashboard
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/licitacoes"
    >
      Licitações
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/SobreLicitacao"
    >
      Sobre as Licitações
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/sobredispensadelicitacao"
    >
      Sobre Dispensas
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/sobrebot"
    >
      Sobre o Bot
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href=""
    >
      Sobre o Projeto
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/sobrenos"
    >
      Sobre Nós
    </a>
  </li>,
  <li
    class="headerListItem"
  >
    <a
      class="headerLink"
      href="/dashboard"
    >
      Dashboard
    </a>
  </li>,
]
`;
